lms/djangoapps/progress/tests.py

```diff
@@ -307,6 +319,37 @@ def test_save_completion_with_course_already_ended(self):
                 content_id=self.problem4.location
             )
 
+    @override_settings(PROGRESS_DETACHED_CATEGORIES=["group-project"])
+    def test_progress_calc_on_invalid_module(self):
+        """
+        Tests progress calculations for invalid modules
+        """
+        self._create_course()
+        # create a module whose parent is not a vertical
+        module = ItemFactory.create(
+            parent_location=self.chapter1.location,
+            category='video',
+            data={'data': '<video display_name="Test Video" />'}
+        )
+        module = self.get_module_for_user(self.user, self.course, module)
+        module.system.publish(module, 'progress', {})
+
+        progress = StudentProgress.objects.all()
+        # assert there is no progress entry for a module whose parent is not a vertical
+        self.assertEqual(len(progress), 0)
+
+        # create a module whose category is one of detached categories
```

Comment:
Nitty-nit-nit: Split this into two test cases -- one for the vertical scenario, and one for the detached category scenario